add essential parameters to sendtoaddress command

debreczeni opened the pull request:
Helps fine-tuning against transaction malleability and also allows to change the default fee which is important with the recent fully crowded blocks during busy hours.

My aim was to not change default values to make sure we won't be breaking existing implementation.

debreczeni commented:
Dear BitGo team,
I have rebased and fixed conflicts so it sits on top of most recent version.
Do you have any thoughts on this pull request?
Many Thanks
David @[MrCoin](https://www.mrcoin.eu) / Shinrai

bencxr commented on src/bgcl.js:
```diff
@@ -472,6 +472,14 @@ BGCL.prototype.createArgumentParser = function() {
   sendToAddress.addArgument(['--confirm'], {action: 'storeConst', constant: 'go', help: 'skip interactive confirm step -- be careful!'});
 
+  sendToAddress.addArgument(['--fee'], {help: 'optional - the fee to use with this transaction.  if not provided, a default, minimum fee will be used'});
+  sendToAddress.addArgument(['--feeRate'], {help: 'optional - the amount of fee per kilobyte - specify either fee, feeRate, or feeTxConfirmTarget but not more than one'});
```
could you use something like: {type: 'int', nargs: '?', help: 'optional - the amount of fee per kilobyte - specify either fee, feeRate, or feeTxConfirmTarget but not more than one'});

this will check the type to be a int and be optional

bencxr commented on src/bgcl.js:
```diff
@@ -1846,10 +1854,20 @@ BGCL.prototype.handleSendToAddress = function() {
       changeAddress: wallet.id()
     };
 
+    for (let optionalParam of ['fee', 'feeRate', 'feeTxConfirmTarget', 'maxFeeRate', 'minConfirms', 'minUnspentsTarget']) {
```
if you can use the type checking in the arg parser above then you can now do a lodash pick + extend txParams.

debreczeni commented:
Hi @bencxr,
Can you please check if my new commit is in line with what you have proposed?
Cheers
David

debreczeni commented:
Dear BitGo Team, 

Is there any willingness on your side to merge this branch?
I have rebased this branch on the latest commit and cleaned the commits a bit.

Thank you.

Cheers,
David

cc @bencxr @jlopp @bpdavenport

bencxr commented:
Hi @debreczeni thanks for submitting this PR!
I like the new inclusion of command line param options but please do not change the sendtoaddress - would prefer sendmany command (more in line with typical crypto libs)

debreczeni commented:
Hi @bencxr 
Thanks for the suggestion.
Actually the "send to many" feature was supposed to go on a separate branch, I have now fixed that.
The original pull request I opened a year ago could then be merged without any major change.
Thinking forward on putting the send to many feature in a separate command - do you suggest copying the whole command and only change the amount/destination and "multi" recipient params?
because everything else looks to be the same...
Or do you have any idea where else they would differ? To me they look all the same, in fact technically the send to address does nothing more than restrict the destination and amount to one recipient, and passes the recipients array for transactionBuilder.CreateTransaction with only one element.
Cheers,
David

debreczeni commented:
closed in favor of #9
